src/redux/reducers/films.ts

```diff
@@ -31,10 +31,37 @@ export const filmsSlice = createSlice({
     },
     setSelectedSortOption: (state, action: PayloadAction<number>) => {
       state.selectedSortOption = action.payload
+    },
+    updateFilm: (state, action: PayloadAction<FilmData>) => {
+      const { payload } = action
+      const filmsCopy = [...state.films]
+      const filmInd = filmsCopy.findIndex((it) => it.id === payload.id)
+      if (filmInd !== -1) {
+        filmsCopy.splice(filmInd, 1, payload)
```

Comment:
it's better to use direct insert by id `filmsCopy[filmInd] = payload` for better readability and micro optimization